features/storage/nvstore/source/nvstore.cpp

```diff
@@ -169,10 +169,35 @@ uint16_t NVStore::get_max_possible_keys()
     return (uint16_t)std::min(max_possible_keys, (size_t) last_reserved_key);
 }
 
-void NVStore::set_max_keys(uint16_t num_keys)
+int NVStore::set_max_keys(uint16_t num_keys)
```

Comment:
This is a breaking change. Although it is correct, not sure we want to change the API.